server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1106,8 +1106,8 @@ private ReplicationGroup calculateReplicationGroup() {
         } else {
             newVersion = replicationGroup.getVersion() + 1;
         }
-        assert indexSettings.isRemoteTranslogStoreEnabled()
-            // Handle migration cases. Ignore assertion if any of the shard copies in the replication group is assigned to a remote node
+        assert newVersion == 0 || indexSettings.isRemoteTranslogStoreEnabled()
+        // Handle migration cases. Ignore assertion if any of the shard copies in the replication group is assigned to a remote node
             || (replicationGroup != null
```

Comment:
We don't need this `null` check anymore here. `newVersion` can only be `0` when `replicationGroup` is `null`

https://github.com/opensearch-project/OpenSearch/blob/3c8eafdb7d59647b169e8dd9e5b00e64b82a01e6/server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java#L1104-L1106

Reply:
Removed the check